haicrypt/hcrypt.c

```diff
@@ -41,18 +41,16 @@ int HaiCrypt_SetLogLevel(int level, int logfa)
 
 #if ENABLE_HAICRYPT_LOGGING
 
-static const char* DumpCfgFlags(int flags)
+static const char* DumpCfgFlags(int flags, char* buf)
```

Comment:
The buffer length needs to be passed as well and checked for buffer overruns on buffer access.

Reply:
This is not a public function, and it's predicted to be used only for logging purposes, and logging in this FA must be turned on explicitly by a distinct option (that is, this function will not be used, unless someone makes a special build for development purposes only, with a special option, that is also warned about security concerns when used).

Reply:
Ok, as this is shared with #359, the problem will be solved there, and this PR remerged after this fix, so this problem should disappear.